src/main/scala/nl/biopet/tools/vcfstats/VcfStatsSpark.scala

```diff
@@ -0,0 +1,402 @@
+
+    vcfRecords.unpersist()
+
+    val future = Future.sequence(
+      sampleCompare._1.toList ::: generalStats._1.toList ::: sampleDistribution._1.toList :::
+        genotypeStats._1.toList ::: infoCounts.flatMap(_._2._1).toList ::: genotypeCounts.flatMap(_._2._1).toList)
+
+    ChunkResult(future, generalStats._2, genotypeStats._2, sampleDistribution._2, sampleCompare._2,
+      infoCounts.map(x => x._1 -> x._2._2), genotypeCounts.map(x => x._1 -> x._2._2))
+  }
+
+  /** Returns the contig dir of a given contig */
+  def contigDir(outputDir: File, contig: String) =
+    new File(outputDir, "contigs" + File.separator + contig)
+
+  /** This method will load a list of regions into memory */
+  def loadVcfFile(cmdArgs: Broadcast[Args],
+                  regions: Broadcast[List[BedRecord]],
+                  prefixMessage: String)(
+                   implicit sc: SparkContext): RDD[VariantContext] = {
+    sc.setJobGroup(s"$prefixMessage Loading Vcf Records", s"$prefixMessage Loading Vcf Records")
+    val vcfRecords = spark.vcf
+      .loadRecords(cmdArgs.value.inputFile, regions, cmdArgs.value.binSize, cached = true)
+    vcfRecords.countAsync()
+    sc.clearJobGroup()
+    vcfRecords
+  }
+
+  /** This will generate contigs stats and write this if this is not disabled */
+  def contigGeneralStats(vcfRecords: RDD[VariantContext],
+                         cmdArgs: Broadcast[Args],
+                         regions: Broadcast[List[BedRecord]],
+                         prefixMessage: String): (Option[FutureAction[Unit]], Option[RDD[(String, GeneralStats)]]) = {
+    if (cmdArgs.value.skipGeneral) (None, None)
+    else {
+      vcfRecords.sparkContext.setJobGroup(s"$prefixMessage General stats", s"$prefixMessage General stats")
+      val generalContig = spark.vcf.generalStats(vcfRecords, regions).cache()
+
+      val contigFutures = if (!cmdArgs.value.notWriteContigStats) {
+        Some(generalContig.foreachAsync {
+          case (contig, stats) =>
+            val dir = contigDir(cmdArgs.value.outputDir, contig)
+            dir.mkdirs()
+            stats.writeToTsv(new File(dir, "general.tsv"))
+        })
+      } else None
+      vcfRecords.sparkContext.clearJobGroup()
+      (contigFutures, Some(generalContig))
+    }
+  }
+
+  /** This will combine contigs stats and write the output file */
+  def totalGeneralStats(generalContig: List[RDD[(String, GeneralStats)]],
+                        cmdArgs: Broadcast[Args]): Option[Future[Unit]] = {
+    if (generalContig.nonEmpty) {
+      val sc = generalContig.head.context
+      sc.setJobGroup("Total General stats", "Total General stats")
+      val emptyRdd = sc.parallelize(
+        List("total" -> new GeneralStats()))
+      Some(sc.union(emptyRdd :: generalContig)
+        .map("total" -> _._2)
+        .reduceByKey(_ += _)
+        .foreachAsync(
+          _._2.writeToTsv(new File(cmdArgs.value.outputDir, "general.tsv"))))
```

Comment:
There is some use of  `._2` in the code which makes it hard to read in some places. Is it possible to improve this?

Reply:
All I can do is write it like this: 
```scala

        .map { case (_, stats) => "total" -> stats }
        .reduceByKey { case (a,b) => a += b }
        .foreachAsync { case (_,stats) =>
          stats.writeToTsv(new File(cmdArgs.value.outputDir, "general.tsv"))})
```

Not so sure if this improves it. For the reduce for sure not, there _ += _ is more clear. Once the method becomes more complex doing it this way, then it also support multiple lines (because of the `{}`).